frontend/src/pages/Lobby/Lobby.js

```diff
@@ -5,72 +5,130 @@ import ListItem from "@mui/material/ListItem";
   const videoElement = useRef(null);
   const [message, setMessage] = useState("");
-  const [participantStream, setParticipantStream] = useState(localStream);
+  const [userConsent, setUserConsent] = useState(false);
+  const [participantStream, setParticipantStream] = useState(null);
+  const [connectionState, setConnectionState] = useState(null);
   const currentSession = useAppSelector(selectCurrentSession);
   const [searchParams, setSearchParams] = useSearchParams();
+  const [connectedParticipants, setConnectedParticipants] = useState([]);
   const sessionIdParam = searchParams.get("sessionId");
   const participantIdParam = searchParams.get("participantId");
-  console.log(sessionIdParam);
 
   useEffect(() => {
     if (connection && connectionState === ConnectionState.CONNECTED) {
       onGetSession(sessionIdParam);
     }
   }, [connection, connectionState, sessionIdParam]);
-
+  const connectedPeersChangeHandler = async (peers) => {
+    console.groupCollapsed(
+      "%cConnection peer streams change Handler",
+      "color:blue"
+    );
+    console.groupEnd();
+    setConnectedParticipants(peers);
+  };
+  useEffect(() => {
+    connection.on("remoteStreamChange", streamChangeHandler);
+    connection.on("connectionStateChange", stateChangeHandler);
+    connection.on("connectedPeersChange", connectedPeersChangeHandler);
+    return () => {
+      // Remove event handlers when component is deconstructed
+      connection.off("remoteStreamChange", streamChangeHandler);
+      connection.off("connectionStateChange", stateChangeHandler);
+      connection.off("connectedPeersChange", connectedPeersChangeHandler);
+    };
+  }, [connection]);
   useEffect(() => {
-    setParticipantStream(localStream);
-  }, [localStream]);
+    if (userConsent) {
+      setParticipantStream(localStream);
+    }
+  }, [localStream, userConsent]);
 
   useEffect(() => {
-    if (participantStream) {
+    if (participantStream && userConsent) {
       videoElement.current.srcObject = localStream;
     }
-  }, [participantStream]);
+  }, [participantStream, userConsent]);
 
   const TabText = styled(Typography)(({ theme }) => ({
     color: theme.palette.primary.main
   }));
 
+  const streamChangeHandler = async () => {
+    console.log("%cRemote Stream Change Handler", "color:blue");
+  };
+  /** Handle `connectionStateChange` event of {@link Connection} */
+  const stateChangeHandler = async (state) => {
+    setConnectionState(state);
+  };
+
+  /** Get the title displayed in a {@link Video} element for `peer`. */
+  const getVideoTitle = (peer: ConnectedPeer, index: number) => {
+    if (peer.summary) {
+      if (peer.summary instanceof Object) {
+        return `${peer.summary.participant_name}`;
+      }
+      return `UserID: ${peer.summary}`;
+    }
+    return `Peer stream ${index + 1}`;
+  };
   return (
     <>
       <AppToolbar />
-      <ConsentModal />
+      <ConsentModal onConsentGiven={setUserConsent} />
```

Comment:
If we use a separate page for consent, we can remove the modal, but if not, it looks good.

Reply:
I think that will be updated with Berke's pr so I will just leave this like this